Diagnostics/HealthChecker/Analyzer/Invoke-AnalyzerEngine.ps1

```diff
@@ -399,52 +399,80 @@
             -AnalyzedInformation $analyzedResults
     }
 
-    $displayWriteType2012 = "Yellow"
-    $displayWriteType2013 = "Yellow"
-    $displayValue2012 = "Unknown"
-    $displayValue2013 = "Unknown"
-
     if ($null -ne $osInformation.VcRedistributable) {
         Write-VerboseOutput("VCRedist2012 Testing value: {0}" -f [HealthChecker.VCRedistVersion]::VCRedist2012.value__)
         Write-VerboseOutput("VCRedist2013 Testing value: {0}" -f [HealthChecker.VCRedistVersion]::VCRedist2013.value__)
-        $vc2013Required = $exchangeInformation.BuildInformation.ServerRole -ne [HealthChecker.ExchangeServerRole]::Edge
-        $displayValue2012 = "Redistributable is outdated"
-        $displayValue2013 = "Redistributable is outdated"
 
         foreach ($detectedVisualRedistVersion in $osInformation.VcRedistributable) {
             Write-VerboseOutput("Testing {0} version id '{1}'" -f $detectedVisualRedistVersion.DisplayName, $detectedVisualRedistVersion.VersionIdentifier)
 
-            if ($detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2012) {
-                $displayValue2012 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
-                $displayWriteType2012 = "Green"
-            } elseif ($vc2013Required -and
-                $detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2013) {
-                $displayWriteType2013 = "Green"
-                $displayValue2013 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
+            if ($detectedVisualRedistVersion.DisplayName -like "Microsoft Visual C++ 2012*") {
+                $vcRedist2012Detected = $true
+                if ($detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2012) {
+                    $displayWriteType2012 = "Green"
+                    $displayValue2012 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
+                } else {
+                    $displayWriteType2012 = "Yellow"
```

Comment:
In my lab, i have multiple `Microsoft Visual C++ 2012` with deferent `VersionIdentifier` this will result incorrect results if the last 2012 found isn't the 'latest' version installed. 

```
[PS] C:\Windows\system32>Get-VisualCRedistributableVersion | group VersionIdentifier

Count Name                      Group
----- ----                      -----
    2 184610406                 {@{DisplayName=Microsoft Visual C++ 2012 x64 Additional Runtime - 11.0.61030; DisplayVersion=11.0.61030; InstallDate=20191203; VersionIdentifier=184610406}, @{DisplayName=Microsoft Visual C++ 2012 x64 ...
    2 201347597                 {@{DisplayName=Microsoft Visual C++ 2013 x64 Additional Runtime - 12.0.21005; DisplayVersion=12.0.21005; InstallDate=20190124; VersionIdentifier=201347597}, @{DisplayName=Microsoft Visual C++ 2013 x64 ...
    2 184600103                 {@{DisplayName=Microsoft Visual C++ 2012 x64 Minimum Runtime - 11.0.50727; DisplayVersion=11.0.50727; InstallDate=20191203; VersionIdentifier=184600103}, @{DisplayName=Microsoft Visual C++ 2012 x64 Add...
```

Reply:
I remember... that was the reason we solved it this way. I'll check in a new change. Please validate in your lab.